source/common/router/retry_state_impl.cc

```diff
@@ -281,10 +281,6 @@ RetryStatus RetryStateImpl::shouldHedgeRetryPerTryTimeout(DoRetryCallback callba
 }
 
 bool RetryStateImpl::wouldRetryFromHeaders(const Http::ResponseHeaderMap& response_headers) {
-  if (response_headers.EnvoyOverloaded() != nullptr) {
```

Comment:
Can you remove this from the O(1) header map? I think we don't need this anymore as the header can be added by string reference where it is added currently?